### PR TITLE
Removed Hardcoded paths

### PR DESCRIPTION
**Pull request by stealthanthrax**
#237 

Removed hardcoded paths from the repo

A definitive path is difficult to remove from `99-susi-usb-drive.rules` file as it will not be executed as a script. But what can be done is instead of using `/home/pi/SUSI.AI` , `~/SUSI.AI` can be used which will eliminate the factor of having a user named `pi` but will still restrict the user to make a `SUSI.AI` folder in the home directory

### CODE REVIEW

**Comment by woshikie** [comment]
>A definitive path is difficult to remove from 99-susi-usb-drive.rules file as it will not be executed as a script. But what can be done is instead of using /home/pi/SUSI.AI , ~/SUSI.AI can be used which will eliminate the factor of having a user named pi but will still restrict the user to make a SUSI.AI folder in the home directory

What can also be done is to **not** have a `99-susi-usb-drive.rules` file, but instead create it directly under `/etc/udev/rules.d/` using `media_udev_rule.sh` by removing:
https://github.com/fossasia/susi_linux/blob/34fc1320606d2f16cd02fd9b2132ba5508520fbb/media_daemon/media_udev_rule.sh#L9-L10
and replace those lines with bash commands to create `99-susi-usb-drive.rules` file inside `/etc/udev/rules.d/`. 
Which is definitely doable. e.g. `touch /etc/udev/rules.d/99-susi-usb-drive.rules` will already create a file there.

<details open>
    <summary>Hint</summary>
    You already have the paths here

  
  https://github.com/fossasia/susi_linux/blob/34fc1320606d2f16cd02fd9b2132ba5508520fbb/media_daemon/media_udev_rule.sh#L6-L7

you can make use of these variables and `echo foobar > foo/bar` to add a line, `foobar`  to `bar` in `foo` directory.

</details>

**Comment by hongquan** [comment]
No @woshikie, there should be a separate _99-susi-usb-drive.rules_ file. It can be served as "back up" and when the installed file is lost by accident, this file can be used for restoring.

**Comment by hongquan** [comment]
Many software is also packaged like that, the configuration files are deployed as separate files, not generated by some script. Here is example of _usb-modeswitch_, a software to switch 3G USB dongle from "CD disk" mode to "3G modem" mode:

```sh
$ dpkg -L usb-modeswitch-data                                     
/.
/etc
/etc/usb_modeswitch.d
/lib
/lib/udev
/lib/udev/rules.d
/lib/udev/rules.d/40-usb_modeswitch.rules
/usr
/usr/share
/usr/share/bug
/usr/share/bug/usb-modeswitch-data
/usr/share/bug/usb-modeswitch-data/presubj
/usr/share/doc
/usr/share/doc/usb-modeswitch-data
/usr/share/doc/usb-modeswitch-data/NEWS.Debian.gz
/usr/share/doc/usb-modeswitch-data/README.Debian
/usr/share/doc/usb-modeswitch-data/README.gz
/usr/share/doc/usb-modeswitch-data/changelog.Debian.gz
/usr/share/doc/usb-modeswitch-data/copyright
/usr/share/lintian
/usr/share/lintian/overrides
/usr/share/lintian/overrides/usb-modeswitch-data
/usr/share/usb_modeswitch
/usr/share/usb_modeswitch/configPack.tar.gz
```

File: access_point/server/server.py [hunk 1]
```diff
@@ -13,19 +17,19 @@ def install():
 
 @app.route('/config/<stt>/<tts>/<hotword>/<wake>')
 def config(stt, tts, hotword, wake):
-    subprocess.call(['sudo', 'bash' , 'server/config.sh', stt, tts, hotword, wake])  #nosec #pylint-disable type: ignore
+    subprocess.call(['sudo', 'bash' , server_folder + '/config.sh', stt, tts, hotword, wake])  #nosec #pylint-disable type: ignore
```
Comment (hongquan) [inline access_point/server/server.py]: Should build path with `os.path.join`, not `+` operator.

File: access_point/server/config.sh [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+SCRIPT_PATH=$(realpath $0)
+DIR_PATH=$(dirname $SCRIPT_PATH)
+
 STT="$1"
 TTS="$2"
 HOTWORD="$3"
 WAKE="$4"
 
-cd /home/pi/SUSI.AI/susi_linux
+cd $DIR_PATH/../../
```
Comment (hongquan) [inline access_point/server/config.sh]: Should find a better way than "cd" around!

Comment (Orbiter) [reply access_point/server/config.sh]: maybe true but in the context of atomic changes step by step it is valid to leave that thought out of this commit.

File: access_point/server/server.py (hunk 1 above)
Comment (Orbiter) [reply access_point/server/server.py]: `os.path.join` may be better if we would expect that this script must work in non-linux (i.e. Windows) environments. But that is not the case. Therefore I think its not too bad to just use '+'

**Pull request merged**